app/main/views.py

```diff
@@ -120,10 +131,10 @@ def update(service_id, section):
         )
 
     if form.errors:
+        service.update(form.dirty_data)
```

Comment:
@allait Yes, that's exactly how it works. And yes again—trying to keep that logic out of the template was the idea.